pilot/pkg/config/kube/crd/conversion.go

```diff
@@ -111,6 +111,9 @@ func ResourceName(s string) string {
 
 // ResourceGroup generates the k8s API group for each schema.
 func ResourceGroup(schema *model.ProtoSchema) string {
+	if schema.GroupDomain != "" {
+		return schema.Group + "." + schema.GroupDomain
```

Comment:
Why not add the dot to the actual value of `schema.GroupDomain` to keep that consistent with `model.IstioAPIGroupDomain`, which contains a dot?

Reply:
Those come from parameters elsewhere in the code, it would be easier if those areas only had to configure the domain aspect and not have to include the '.'

Reply:
For example
`		Group:         "authentication",`
`		Version:       "v1alpha1",`
`		GroupDomain:   "maistra.io",`

Reply:
OK. I found this odd because of the comment here: https://github.com/Maistra/istio/blob/5c495dfab939aa419dc0863c57bcb14f59dcbdfe/pilot/pkg/model/config.go#L218

Strictly reading the comment: if it truly defaults to IstioAPIGroupDomain, then it's value contains a dot. But if users set it, they should not add the dot, which is a minor inconsistency.

Reply:
I mean. A user may go look for the value of IstioAPIGroupDomain and think they need to add the dot. And the code here doesn't skip adding the dot if the dot is there.

Reply:
I'll tidy that up

Reply:
I've removed the `.` from the default and am explicitly adding it.